### PR TITLE
Don't try to set Kerberos extradata when there is no principal

### PR DESCRIPTION
**Pull request by rcritten**
This was causing ns-slapd to segfault in the password plugin.

https://pagure.io/freeipa/issue/7561

Signed-off-by: Rob Crittenden <rcritten@redhat.com>

### CODE REVIEW

**Comment by rcritten** [comment]
Working on an integration test.

**Comment by abbra** [comment]
Changes look good to me. I restarted tests as one of them failed to publish results.

File: ipatests/test_integration/test_commands.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+from ipatests.test_integration.base import IntegrationTest
+from ipatests.pytest_plugins.integration import tasks
+
+
+class TestCommands(IntegrationTest):
+    """
+    A lot of commands can be executed against a single IPA installation
+    so provide a generic class to execute one-off commands that need to be
+    tested without having to fire up a full server to run one command.
```
Comment (tiran) [inline ipatests/test_integration/test_commands.py]: I like the idea! :+1: 

I even like the idea so much, that I fired up my time machine and added ``test_ipa_cli`` in the past. :) But your file name better. Please merge and fold ``test_ipa_cli`` content into your new file and update ``.freeipa-pr-ci.yaml``.

File: ipatests/test_integration/test_commands.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        tasks.install_master(cls.master)
+
+    def test_change_sysaccount_password(self):
+        sysuser = 'system'
+        original_passwd = 'Secret123'
+        new_passwd = 'userPasswd123'
+
+        master = self.master
+
+        base_dn = "dc=%s" % (",dc=".join(master.domain.name.split(".")))
```
Comment (tiran) [inline ipatests/test_integration/test_commands.py]: ``str(master.domain.basedn)``?

File: ipatests/test_integration/test_commands.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        ldif_file = tf.name
+        entry_ldif = textwrap.dedent("""
+            dn: uid=system,cn=sysaccounts,cn=etc,{base_dn}
+            changetype: add
+            objectclass: account
+            objectclass: simplesecurityobject
+            uid: system
+            userPassword: {original_passwd}
+            passwordExpirationTime: 20380119031407Z
+            nsIdleTimeout: 0""").format(
```
Comment (tiran) [inline ipatests/test_integration/test_commands.py]: style nit pick. I'd format the line like

```
        nsIdleTimeout: 0
    """).format(
        base_dn=...
```

File: ipatests/test_integration/test_commands.py (hunk 1 above)
Comment (rcritten) [reply ipatests/test_integration/test_commands.py]: Done.

**Comment by tiran** [comment]
master:

* 45d776a7bf05f3495dee078c7dd58ed0db13f64a Don't try to set Kerberos extradata when there is no principal
* 7c5ecb8d08827a60f77eda0911a6b39db5badf82 Rename test class for testing simple commands, add test

**Pull request closed without merging**